.github/workflows/steam.yml

```diff
@@ -0,0 +1,212 @@
+        echo "WIN_ARTIFACT_URL=${WIN_ASSET_URL}" >> $GITHUB_ENV
+        echo "MAC_x86_ARTIFACT_URL=${MAC_ASSET_URL}" >> $GITHUB_ENV
+
+    - name: Restore build cache
+      id: cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/steam/build
+        key: ${{ env.BRANCH }}-${{ env.DESC }}
+        # Using "restore-keys" will restore the most recent cache for the branch, even if the exact cache doesn't exist.
+        # This doesn't set cache-hit to true so it won't skip the upload for nightlies.
+        restore-keys: ${{ env.BRANCH }}
+
+    - name: Determine if Steam upload should run
+      # If the nightly build has already been uploaded and thus a cache exists skip this and the following steps.
+      # Steam does not prevent us from uploading duplicate builds so this would just pollute the dashboard.
+      # This is a bit of a hack and can fail to work if our cache has been evicted or we somehow have no commits for 7 days,
+      # but it's better than nothing!
+      id: should-run
+      run: |
+        SHOULD_RUN=$(if [[ "${{ env.UPLOAD_TYPE }}" == "release" || "${{ steps.cache.outputs.cache-hit }}" != "true" ]]; then echo "true"; else echo "false"; fi)
+        echo "::set-output name=result::$SHOULD_RUN"
+    
+    - name: Download and prepare builds
+      if: steps.should-run.outputs.result == 'true'
+      run: |
+        echo "::group:: Download Windows build"
+        if [[ "${{ env.WIN_ARTIFACT_URL }}" == *"api.github.com"* ]]; then
+            wget --header="Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "${{ env.WIN_ARTIFACT_URL }}" -O windows.zip
+        else
+            wget "${{ env.WIN_ARTIFACT_URL }}" -O windows.zip
+        fi
+        echo "::endgroup::"
+      
+        echo "::group:: Download Mac build"
+        if [[ "${{ env.MAC_x86_ARTIFACT_URL }}" == *"api.github.com"* ]]; then
+            wget --header="Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "${{ env.MAC_x86_ARTIFACT_URL }}" -O mac_x86.dmg.zip
+        else
+            wget "${{ env.MAC_x86_ARTIFACT_URL }}" -O mac_x86.dmg
+        fi
+        echo "::endgroup::"
+
+        mkdir -p steam && cd steam
+  
+        echo "::group::Extract and prepare Win64"
+        mkdir steam-windows
+        (
+            cd steam-windows
+            unzip ../../windows.zip
+            # CI builds can be double-zipped
+            if compgen -G "*.zip" > /dev/null; then
+                unzip *.zip
+                rm *.zip
+            fi
+            # copy install scripts and create sentinel file
+            cp -r ../../source/CI/steam/scripts scripts
+            touch disable_updater
+        )
+        echo "::endgroup::"
+
+        echo "::group::Extract macOS (x86)"
+        mkdir steam-macos
+        # CI builds are zipped
+        if [[ -f ../mac_x86.dmg.zip ]]; then
+            unzip ../mac_x86.dmg.zip
+            # ignore exit code from 7zip
+            7zz x *.dmg -otmp || true
```

Comment:
That is indeed the correct way to "capture" non-0 return codes and avoid a script failure. (Either the or capturing it in a variable for later handling).